CRM/Member/Form/Task.php

```diff
@@ -95,8 +95,11 @@ public static function preProcessCommon(&$form, $useTable = FALSE) {
     $values = $form->controller->exportValues($form->get('searchFormName'));
 
     $form->_task = $values['task'];
-    $memberTasks = CRM_Member_Task::tasks();
-    $form->assign('taskName', $memberTasks[$form->_task]);
+    $tasks = CRM_Member_Task::permissionedTaskTitles(CRM_Core_Permission::getPermission());
+    if (!array_key_exists($form->_task, $tasks)) {
+      CRM_Core_Error::statusBounce(ts('You do not have permission to access this page.'));
```

Comment:
I feel a little nervous about this standardisation  - does this bounce happen BEFORE or AFTER hooks get to 'add their say'

Reply:
@eileenmcnaughton This has been standardised across all tasks.  Previously I think some did this, some did not.  It is on the form layer AFTER hooks have run (the searchTasks hook gets run via permissionedTaskTitles in this case so you are only likely to get here with an invalid task ID if you've manipulated a URL or done something else weird - it was then somewhat undefined as to what would happen, a task might load or you might get a fatal.